pkg/pipeline/pipeline.go

```diff
@@ -600,32 +601,37 @@ func (s AthenaConfig) MarshalJSON() ([]byte, error) {
-	CustomChecks    []CustomCheck      `json:"custom_checks" yaml:"custom_checks,omitempty" mapstructure:"custom_checks"`
-	Metadata        EmptyStringMap     `json:"metadata" yaml:"metadata,omitempty" mapstructure:"metadata"`
-	Snowflake       SnowflakeConfig    `json:"snowflake" yaml:"snowflake,omitempty" mapstructure:"snowflake"`
-	Athena          AthenaConfig       `json:"athena" yaml:"athena,omitempty" mapstructure:"athena"`
-
-	upstream   []*Asset
-	downstream []*Asset
+	ID                string             `json:"id" yaml:"-" mapstructure:"-"`
+	URI               string             `json:"uri" yaml:"uri,omitempty" mapstructure:"uri"`
+	Name              string             `json:"name" yaml:"name,omitempty" mapstructure:"name"`
+	Type              AssetType          `json:"type" yaml:"type,omitempty" mapstructure:"type"`
+	Description       string             `json:"description" yaml:"description,omitempty" mapstructure:"description"`
+	Connection        string             `json:"connection" yaml:"connection,omitempty" mapstructure:"connection"`
+	Tags              EmptyStringArray   `json:"tags" yaml:"tags,omitempty" mapstructure:"tags"`
+	Materialization   Materialization    `json:"materialization" yaml:"materialization,omitempty" mapstructure:"materialization"`
+	Upstreams         []Upstream         `json:"upstreams" yaml:"depends,omitempty" mapstructure:"depends"`
+	Image             string             `json:"image" yaml:"image,omitempty" mapstructure:"image"`
+	Instance          string             `json:"instance" yaml:"instance,omitempty" mapstructure:"instance"`
+	Owner             string             `json:"owner" yaml:"owner,omitempty" mapstructure:"owner"`
+	ExecutableFile    ExecutableFile     `json:"executable_file" yaml:"-" mapstructure:"-"`
+	DefinitionFile    TaskDefinitionFile `json:"definition_file" yaml:"-" mapstructure:"-"`
+	Parameters        EmptyStringMap     `json:"parameters" yaml:"parameters,omitempty" mapstructure:"parameters"`
+	Secrets           []SecretMapping    `json:"secrets" yaml:"secrets,omitempty" mapstructure:"secrets"`
+	Extends           []string           `json:"extends" yaml:"extends,omitempty" mapstructure:"extends"`
+	Columns           []Column           `json:"columns" yaml:"columns,omitempty" mapstructure:"columns"`
+	CustomChecks      []CustomCheck      `json:"custom_checks" yaml:"custom_checks,omitempty" mapstructure:"custom_checks"`
+	Metadata          EmptyStringMap     `json:"metadata" yaml:"metadata,omitempty" mapstructure:"metadata"`
+	Snowflake         SnowflakeConfig    `json:"snowflake" yaml:"snowflake,omitempty" mapstructure:"snowflake"`
+	Athena            AthenaConfig       `json:"athena" yaml:"athena,omitempty" mapstructure:"athena"`
+	upstream          []*Asset
+	downstream        []*Asset
+	IntervalModifiers IntervalModifiers `json:"interval_modifiers" yaml:"interval_modifiers,omitempty" mapstructure:"interval_modifiers"`
```

Comment:
the exported fields have been grouped together at the top, then comes the unexported fields like `upstream` and `downstream`, I suggest grouping them together.